File: src/main/java/tech/ccat/debugger/DebugStatProvider.java
package tech.ccat.debugger;

import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import tech.ccat.kstats.api.StatProvider;
import tech.ccat.kstats.model.PlayerStat;
import tech.ccat.kstats.model.StatType;

import java.util.Collections;
import java.util.Map;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

public class DebugStatProvider implements StatProvider {
    // 存储玩家调试数值：UUID -> (属性类型 -> 调整值)
    private final ConcurrentHashMap<UUID, Map<StatType, Double>> debugValues = new ConcurrentHashMap<>();

    public void adjustStat(Player player, StatType statType, double amount) {
        Map<StatType, Double> playerData = debugValues.computeIfAbsent(
                player.getUniqueId(),
                k -> new ConcurrentHashMap<>()
        );
        playerData.merge(statType, amount, Double::sum);
    }

    @NotNull
    @Override
    public PlayerStat provideStats(Player player) {
        Map<StatType, Double> adjustments = debugValues.get(player.getUniqueId());
        final Map<StatType, Double> stats = adjustments != null ? adjustments : Collections.emptyMap();

        return new PlayerStat(
                stats.getOrDefault(StatType.HEALTH, 0.0),
                stats.getOrDefault(StatType.DEFENSE, 0.0),
                stats.getOrDefault(StatType.STRENGTH, 0.0),
                stats.getOrDefault(StatType.SPEED, 0.0),
                stats.getOrDefault(StatType.CRIT_CHANCE, 0.0),
                stats.getOrDefault(StatType.CRIT_DAMAGE, 0.0),
                stats.getOrDefault(StatType.WISDOM, 0.0),
                stats.getOrDefault(StatType.BASE_DAMAGE, 0.0),
                stats.getOrDefault(StatType.DAMAGE_MULTIPLIER, 0.0),
                stats.getOrDefault(StatType.HEALING, 0.0)
        );
    }
}

File: src/main/kotlin/tech/ccat/debugger/DebugStatProvider.kt
//// tech/ccat/debugger/DebugStatProvider.kt
//package tech.ccat.debugger
//
//import org.bukkit.entity.Player
//import tech.ccat.kstats.api.StatProvider
//import tech.ccat.kstats.model.PlayerStat
//import tech.ccat.kstats.model.StatType
//import java.util.*
//import java.util.concurrent.ConcurrentHashMap
//
//class DebugStatProvider : StatProvider {
//    // 存储玩家调试数值：UUID -> (属性类型 -> 调整值)
//    private val debugValues = ConcurrentHashMap<UUID, MutableMap<StatType, Double>>()
//
//    fun adjustStat(player: Player, statType: StatType, amount: Double) {
//        val playerData = debugValues.computeIfAbsent(player.uniqueId) { ConcurrentHashMap() }
//        playerData.merge(statType, amount) { a, b -> a + b }
//    }
//
//    override fun provideStats(player: Player): PlayerStat {
//        return debugValues[player.uniqueId].let { adjustments ->
//            PlayerStat(
//                health = adjustments?.getOrDefault(StatType.HEALTH, 0.0).,
//                defense = adjustments?.getOrDefault(StatType.DEFENSE, 0.0),
//                strength = adjustments?.getOrDefault(StatType.STRENGTH, 0.0),
//                speed = adjustments?.getOrDefault(StatType.SPEED, 0.0),
//                critChance = adjustments?.getOrDefault(StatType.CRIT_CHANCE, 0.0),
//                critDamage = adjustments?.getOrDefault(StatType.CRIT_DAMAGE, 0.0),
//                wisdom = adjustments?.getOrDefault(StatType.WISDOM, 0.0),
//                baseDamage = adjustments?.getOrDefault(StatType.BASE_DAMAGE, 0.0),
//                damageMultiplier = adjustments?.getOrDefault(StatType.DAMAGE_MULTIPLIER, 0.0),
//                healing = adjustments?.getOrDefault(StatType.HEALING, 0.0)
//            )
//        }
//    }
//}